libevmasm/Assembly.cpp

```diff
@@ -1281,8 +1281,11 @@ LinkerObject const& Assembly::assembleLegacy() const
 	uint8_t tagPush = static_cast<uint8_t>(pushInstruction(bytesPerTag));
 	uint8_t dataRefPush = static_cast<uint8_t>(pushInstruction(bytesPerDataRef));
 
+	size_t instructionIndex = 0;
 	for (AssemblyItem const& item: items)
```

Comment:
A more modern version of manually incrementing the instruction index would be to use `ranges::views::enumerate(items)` :)

Amounts to something like `for (auto const& [instructionIndex, item]: enumerate(items)) { ... }`

Reply:
At first I changed that so that `ranges::views::enumerate`but than I thought it would be better readable to introduce that `addInstructionOffset` lambda. It also made defining of that `offsetAfterLastInstruction` a bit ugly, so I decided not to use that here.

Reply:
I think it would be more readable if you made it side-effect free and instead explicitly passed `ret` and `instructionIndex` into the function. The code is easier to understand when you see explicitly what goes in and out.